dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -1466,3 +1466,11 @@ def refute_bad_gateway_or_site_unreachable
   expect(params['name']).to eq(name)
   expect(params['course']).to eq(course)
 end
+
+And(/^I validate rubric ai config for all lessons$/) do
```

Comment:
would this UI test be possible on a development environment without AWS access and/or contributor access?

Reply:
I meant to ask this on the test and not the step 🥲

Reply:
no, this test would fail if you did not have S3 access. I don't think developers are often running all UI tests locally, but I could see how this could be a regression for a contributor working on ai rubric features.

other UI tests depend on S3 though (most labs features I think) so I'm not sure this is a big outlier in this way though.

per other comment, I hope to move this config into aiproxy, but i agree that i shouldn't leave things in a bad state in the interim.

Reply:
perhaps UI tests should be flagged with `@s3_access`, so we can intelligently skip them (or give helpful warning) when a developer without s3 access tries to run them? we do something similar for `@dashboard_db_access`, which has to do with whether you are running the runner.rb on the same machine as the rails server. so the implementation could be similar.

Reply:
That feels like a nice thing to have! I try to run UI tests locally :) I'm part of the eng-excellence team for remove AWS deps so I'm looking for best practices moving forward. Things like skipping AWS dependent tests when we run them locally / don't have a valid key would be nice to have. Could we flag them even if we don't yet implement that check at least?

Reply:
to clarify, I agree that most engineers run individual UI tests locally for the features they are working on, but that it is uncommon for a contributor to work on a feature which does not require S3 access and yet still need to run UI tests which require S3 access. I won't fight you on that overall eng initiative though. I'm happy to store the "releases" somewhere else, but not sure where that might be?

> Things like skipping AWS dependent tests when we run them locally / don't have a valid key would be nice to have. Could we flag them even if we don't yet implement that check at least?

we could start speculatively adding a flag to ui tests that use S3, but without anything checking this, it might be hard to keep this list in sync. I count 25 ui test files containing the string `studio.code.org/projects/`, and that's just one indicator of S3 access. Perhaps a more definitive way to tag them would be to take away S3 access and see which ones fail.

Reply:
I'd say that this shouldn't block merging this... there are already dozens that would be similarly broken for non-AWS access. I generally develop without AWS access, myself, so I do run into these from time to time, but I'm going to advocate we fix that separately!